lib/service/connect.go

```diff
@@ -426,6 +425,15 @@ func (process *TeleportProcess) firstTimeConnect(role types.SystemRole) (*Connec
 	if err != nil {
 		return nil, trace.NewAggregate(err, connector.Close())
 	}
+
+	if role == types.RoleInstance {
+		// Instance always joins first, only try to persist host ID after
+		// successfully completing the instance join and persisting the state
+		// and identity.
+		if err := process.storage.PersistAssignedHostID(process.ExitContext(), process.Config, identity.ID.HostID()); err != nil {
```

Comment:
Should we use `process.GracefulExitContext()` instead of `process.ExitContext()`?

Reply:
switched